File: asr/harmony/sdk/src/main/ets/com/amphion/asr/NativeEndpointTransitionPolicy.ts
export class NativeEndpointTransition {
  static readonly HARD_RESTART = new NativeEndpointTransition('hard-restart');
  static readonly NATIVE_CHECKPOINT = new NativeEndpointTransition('native-checkpoint');

  private readonly value: string;

  private constructor(value: string) {
    this.value = value;
  }
}

export class NativeEndpointTransitionPolicy {
  static decide(isRule3Endpoint: boolean, hasEvidence: boolean,
    speakerVadEnabled: boolean): NativeEndpointTransition {
    return isRule3Endpoint && hasEvidence && !speakerVadEnabled ?
      NativeEndpointTransition.NATIVE_CHECKPOINT : NativeEndpointTransition.HARD_RESTART;
  }
}

File: asr/android/sdk/src/main/java/com/amphion/asr/internal/NativeEndpointTransitionPolicy.kt
package com.amphion.asr.internal

import com.k2fsa.sherpa.onnx.OnlineEndpointReason

internal enum class NativeEndpointTransition {
    HARD_RESTART,
    NATIVE_CHECKPOINT,
}

internal object NativeEndpointTransitionPolicy {
    fun decide(
        reason: OnlineEndpointReason,
        hasEvidence: Boolean,
        isFinalFlush: Boolean,
    ): NativeEndpointTransition =
        if (reason == OnlineEndpointReason.RULE3 && hasEvidence && !isFinalFlush) {
            NativeEndpointTransition.NATIVE_CHECKPOINT
        } else {
            NativeEndpointTransition.HARD_RESTART
        }
}
